backend/src/data/repository/user.ts

```diff
@@ -44,6 +32,43 @@ export class UserRepository {
     return data;
   }
 
+  async getUsers(): Promise<DTO.IUserDoc[] | null> {
+    const data = await User.find(
+      {},
+      { firstName: 1, lastName: 1, photo: 1, birthday: 1, country: 1, speak: 1, learn: 1 }
+    ).limit(5);
+    return data;
+  }
+  // Search
+
+  async getUsersByParams(params: DTO.FilterRequest): Promise<DTO.IUserDoc[] | null> {
+    const { name, lowAge, highAge, sex, country, language, level } = params;
+    const data = await User.find({
+      $and: [
+        {
+          $expr: {
+            $regexMatch: {
+              input: { $concat: ['$firstName', ' ', '$lastName'] },
+              regex: name,
+              options: 'i'
+            }
+          }
+        },
+        {
+          birthday: {
+            $gte: new Date(highAge).toISOString(),
```

Comment:
Are you sure that here should not be `$gte` for `lowAge`?

Reply:
We have birthday "1950-01-01T00:00:00.000Z" if we use gte, it mean 1950-01-02 and bigger,
for lte, we get current year 2020 -  age 20  = ~2000, then we add 12, 31 last day of year and it displays all users whose birthday is lower or equal 2000-12-31